### PR TITLE
express-api-birgit

### PR DESCRIPTION
**Pull request by nehrwein**
Visit my project: https://nehrwein-netflix-api.herokuapp.com/

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -21,13 +14,135 @@ const app = express()
...
+app.get('/endpoints', (req, res) => {
+  res.json({
+    response: listEndpoints(app),
+    success: true
+  })
+})
+
+// route provides a sorted list of all countries that are in netflixData
+app.get('/countries', (req, res) => {
+  const countries = Array.from(new Set(netflixData.map(item => item.country))).sort()
```
Comment (DALA746) [inline server.js:35]: Could you leave a comment about this line of code? :)

File: server.js [hunk 2]
```diff
@@ -21,13 +14,135 @@ const app = express()
...
+  res.json({
+    response: filteredMovies,
+    success: true
+  })
+})
+
+// route provides one movie by ID
+app.get('/movies/id/:id', (req, res) => {
+  const { id } = req.params
+  const movies = netflixData.filter(item => item.type === "Movie")
```
Comment (DALA746) [inline server.js:106]: Here you are filtering by type, maybe rename this variable to something else? Maybe filteredByType?

File: server.js [hunk 3]
```diff
@@ -21,13 +14,135 @@ const app = express()
...
+  }  
+
+  res.json({
+    response: filteredMovies,
+    success: true
+  })
+})
+
+// route provides one movie by ID
+app.get('/movies/id/:id', (req, res) => {
```
Comment (DALA746) [inline server.js:104]: As I understand, this route gives only a movie with a specific id, but why do you filter on line 106? When using params it only returns one specific item as I understand 🤔

Comment (DALA746) [reply server.js:104]: Aaa, now I saw that you are using movies variable on line 108 :) maybe it is a good idea to change variable name :)

File: server.js [hunk 4]
```diff
@@ -21,13 +14,135 @@ const app = express()
...
+      success: true
+    })
+  } 
+})
+
+// route provides all movies and has the possibility to query for director, year and actor. You can also do pagination by setting page & limit as query parameters
+app.get('/movies', (req, res) => {
+  const { director, year, actor, page, limit } = req.query
+
+  const movies = netflixData.filter(item => item.type === "Movie")
```
Comment (DALA746) [inline server.js:68]: You only have movies, not TV shows in your project so that is why you are filtering by type :)

**Review by DALA746 — COMMENTED** [review]
Readable and understandable code to read! Nice that you implemented documentation with Postman! I did my project with the same data so I couldn't add more feedback! You implemented a different approach on line 35 which is great, but leave the comment next time just to explain the code. Nice job! 💯

File: server.js (hunk 1 above)
Comment (nehrwein) [reply server.js:35]: Sure :)
The aim with this code is to show a list of all countries in the Netflix-Data without any duplicates. Therefore I am creating a Set (object with unique items). I convert the set into an array afterwards and sort it alphabetically

File: server.js (hunk 3 above)
Comment (nehrwein) [reply server.js:105]: now I made movies a global variable. I hope that's more clear now and also has the advantage of less repetitive code. Thanks for pointing out!

File: server.js [hunk 5]
```diff
@@ -1,33 +1,147 @@
...
+  res.json({
+    response: listEndpoints(app),
+    success: true
+  })
+})
+
+// route provides a sorted list of all countries that are in netflixData
+app.get('/countries', (req, res) => {
+  // a list of all countries in the Netflix-Data without any duplicates. Therefore I am creating a Set (object with unique items). I convert the set into an array afterwards and sort it alphabetically
+  const countries = Array.from(new Set(netflixData.map((item) => item.country))).sort()
```
Comment (Jsfrulle) [inline server.js:38]: oh good code :D this gives the coutries in alphabetical order i guess

File: server.js [hunk 6]
```diff
@@ -1,33 +1,147 @@
...
+    })
+  } else {
+    res.json({
+      response: movie,
+      success: true
+    })
+  }
+})
+
+// route provides movies by name (can return more than one movie, if the provided parts of the title match with several movies)
```
Comment (Jsfrulle) [inline server.js:123]: really nice way to let the frontender use this array!!! Like if you want to find all film with a word in its title!!!! this is a setup up compare to what i done in my project!

File: server.js (hunk 5 above)
Comment (Jsfrulle) [reply server.js:38]: maybye add a 404 respons

**Pull request closed without merging**